ra/registration-authority_test.go

```diff
@@ -204,7 +225,48 @@ func initAuthorities(t *testing.T) (core.CertificateAuthority, *DummyValidationA
 	AuthzFinal.Expires = &exp
 	AuthzFinal.Challenges[0].Status = "valid"
 
-	return &ca, va, sa, &ra
+	return &ca, va, ssa, &ra, cleanUp
+}
+
+// This is an unfortunate bit of tech debt that is being taken on in
+// order to get the more important change of using MySQL/MariaDB in
+// all of our tests working without SQLite. We already had issues with
+// the RA here getting a real CertificateAuthority instead of a
+// CertificateAuthorityClient, so this is only marginally worse.
+func caDBImpl(t *testing.T) (core.CertificateAuthorityDatabase, func()) {
+	dbMap, err := sa.NewDbMap(dbConnStr)
+	if err != nil {
+		t.Fatalf("Could not construct dbMap: %s", err)
+	}
+
+	cadb, err := ca.NewCertificateAuthorityDatabaseImpl(dbMap)
+	if err != nil {
+		t.Fatalf("Could not construct CA DB: %s", err)
+	}
+
+	// We intentionally call CreateTablesIfNotExists twice before returning
+	// because of the weird insert inside it.
```

Comment:
I added this comment:

```
	// We intentionally call CreateTablesIfNotExists twice before
	// returning because of the weird insert inside it. The
	// CADatabaseImpl code expects the existence of a single row in
	// its serialIds table or else it errors. CreateTablesIfNotExists
	// currently inserts that row and TruncateTables will remove
	// it. But we need to make sure the tables exist before
	// TruncateTables can be called to reset the table. So, two calls
	// to CreateTablesIfNotExists.
```